### PR TITLE
add runto command

### PR DESCRIPTION
**Pull request by chrisza4**
Add `runto` command to run until specific step.

Most of the time, when we setup prescript test with selenium, we wish to skip all the setup step. 

For example, we mostly ended up like this

Test A:

1. Create user
 1.1. Enter input username
 1.2. Enter password
 1.3. Confirm password
 1.4. Click on...
 1.5. .... etc
2. Confirm email
 2.1 Go to email website
 2.2 Login email address
 2.3 .....
3. .....
4. ....
5. Real test start here

Test B: 

1. Create user
 1.1. Enter input username
 1.2. Enter password
 1.3. Confirm password
 1.4. Click on...
 1.5. .... etc
2. Confirm email
 2.1 Go to email website
 2.2 Login email address
 2.3 .....
3. .....
4. ....
5. Real test start here

The `next` functionality have no benefit in these cases. We need command to immediately goto step 5 and then start debugging test from there

### CODE REVIEW

File: lib/cli.js [hunk 1]
```diff
@@ -157,6 +157,21 @@ function runDevelopmentMode (testModulePath) {
     callback()
   })
 
+  vorpal.command('runto <stepNumber>')
+  .description('Run from current step to step number')
+  .action(function (args, callback) {
+    handleRun(co(function* () {
+      while (!(tester.getCurrentStepNumber() === String(args.stepNumber) || tester.isDone())) {
```
Comment (dtinth) [inline lib/cli.js]: This block is duplicated from `continue` command.

Please deduplicate, e.g.:

```js
    continueRunning({
      shouldStop: (status) => {
        return status.nextStepNumber === String(args.stepNumber)
      },
      callback: callback
    })
```

File: lib/cli.js [hunk 2]
```diff
@@ -157,6 +157,21 @@ function runDevelopmentMode (testModulePath) {
     callback()
   })
 
+  vorpal.command('runto <stepNumber>')
+  .description('Run from current step to step number')
+  .action(function (args, callback) {
+    handleRun(co(function* () {
```
Comment (dtinth) [inline lib/cli.js]: There should be a space before `*`, per standard.

This project is using `standard` linter CLI directly, not through ESLint.

Sorry I forgot to make the CI run the linter.

File: lib/cli.js (hunk 1 above)
Comment (dtinth) [inline lib/cli.js]: Another thought:

What if we make it an option for the `continue` command?

e.g.

```
continue --until 5
```

Or in short

```
c -u 5
```

File: lib/cli.js (hunk 2 above)
Comment (dtinth) [inline lib/cli.js]: It should also check if target step exists.

If not, it should refuse to run.

Another edge case here is that if the action step is 5.1, user should be able to say `runto 5` and it should pause at 5.1.

**Comment by chrisza4** [comment]
I think the de-duplication look really weird.

I want to comment that while I'm working on modify the `cli.js`. I think there are too many implicit context exists. We have implicit `tester`, `clearModuleCache`, `loadTest`, `nextStep`

It's really hard to know how things works here. I proposed we should do either one of these options.

1. Add more explicit context into each function, such as adding `tester` as argument
2. Re-organized order of private function and variable context to make it more clear what can be use while developing on this module.
3. Reduce number of context variable or context function exists in this module.

This is feedback/comment that I want to let you know how I feel and think while I'm modifying code here. But there is no need to fix this. We can refactor this later.

File: lib/cli.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ const chalk = require('chalk')
 const indentString = require('indent-string')
 
 const createTestIterator = require('./createTestIterator')
-const walkSteps = require('./walkSteps')
+const { walkSteps, isStepExists } = require('./stepsTraverseUtils')
```
Comment (dtinth) [inline lib/cli.js]: I would use two files.

File: lib/stepsTraverseUtils.js [hunk 4]
```diff
@@ -6,4 +6,15 @@ function walkSteps (root, visit) {
   for (const child of root.children) traverse(child)
 }
 
-module.exports = walkSteps
+function isStepExists (root, number) {
```
Comment (dtinth) [inline lib/stepsTraverseUtils.js]: Exist

**Pull request merged**